srv/salt/ceph/monitoring/prometheus/exporters/ceph_rgw_exporter/default.sls

```diff
@@ -0,0 +1,16 @@
+install_package:
+  pkg.installed:
+    - name: python-prometheus-client
+    - refresh: True
+
+install_rgw_exporter:
+  file.managed:
+    - name: /var/lib/prometheus/node-exporter/ceph_rgw.py
+    - user: prometheus
+    - group: prometheus
+    - mode: 755
+    - source: salt://{{ slspath }}/files/ceph_rgw.py
```

Comment:
Running the DeepSea suite with this PR in OVH, Stage 4 is now failing with:

```
2017-11-23T15:01:48.838 INFO:teuthology.orchestra.run.target167114232165.stderr:+ deepsea --log-file=/var/log/salt/deepsea.log --log-level=debug stage run ceph.stage.4 --simple-output
2017-11-23T15:01:49.076 INFO:teuthology.orchestra.run.target167114232165.stdout:Starting stage: ceph.stage.4
2017-11-23T15:01:59.881 INFO:teuthology.orchestra.run.target167114232165.stdout:Parsing stage ceph.stage.4 steps... fail
2017-11-23T15:01:59.882 INFO:teuthology.orchestra.run.target167114232165.stdout:
2017-11-23T15:01:59.883 INFO:teuthology.orchestra.run.target167114232165.stdout:An error occurred when rendering one of the following states:
2017-11-23T15:01:59.883 INFO:teuthology.orchestra.run.target167114232165.stdout:    - ceph.rgw.auth (/srv/salt/ceph/rgw/auth)
2017-11-23T15:01:59.883 INFO:teuthology.orchestra.run.target167114232165.stdout:    - ceph.rgw.users (/srv/salt/ceph/rgw/users)
2017-11-23T15:01:59.884 INFO:teuthology.orchestra.run.target167114232165.stdout:    - ceph.monitoring.prometheus.exporters.ceph_rgw_exporter (/srv/salt/ceph/monitoring/prometheus/exporters/ceph_rgw_exporter)
2017-11-23T15:01:59.884 INFO:teuthology.orchestra.run.target167114232165.stdout:    - ceph.rgw.buckets (/srv/salt/ceph/rgw/buckets)
2017-11-23T15:01:59.884 INFO:teuthology.orchestra.run.target167114232165.stdout:
2017-11-23T15:01:59.884 INFO:teuthology.orchestra.run.target167114232165.stdout:Error description:
2017-11-23T15:01:59.884 INFO:teuthology.orchestra.run.target167114232165.stdout:SaltRenderError: Jinja variable 'slspath' is undefined
2017-11-23T15:01:59.885 INFO:teuthology.orchestra.run.target167114232165.stdout:
2017-11-23T15:01:59.926 INFO:teuthology.orchestra.run.target167114232165.stderr:+ deepsea_exit_status=2
2017-11-23T15:01:59.927 INFO:teuthology.orchestra.run.target167114232165.stderr:+ echo 'deepsea exit status: 2'
```

I'm still investigating to see how reproducible it is. More to come.

Reply:
@votdev @swiftgist @rjfd This is 100% reproducible using the CLI. When running Stage 4 without the CLI, the failure doesn't happen. I grepped the codebase for "slspath" and it only occurs in this one place. . . so I'm wondering how it's supposed to get defined?

Reply:
Detailed logs here: http://54.37.31.117:8081/ubuntu-2017-11-23_22:24:34-deepsea:basic:rgw-ses5---basic-openstack/

Reply:
I can not reproduce this using Salt 2016.11.4 which is used in SES5 as far as i know.

Reply:
slspath is defined by Salt, see https://docs.saltstack.com/en/2016.11/ref/states/vars.html.

Reply:
I'll check why the CLI is failling to find the `slspath` variable.

Reply:
The problem is in the way how the CLI parses the sls files. It uses an execution modules called `cp.get_template` which does not know the `slspath` variable.

@votdev Since the change of the CLI to overcome this problem might be not trivial, could we just remove the `slspath` variable and substitute it with an hardcoded path as it is done in all the remaining state files of the project?

I'll open a new issue for the CLI nevertheless.

Reply:
@rjfd No problem, i'll create a new PR which will replace slspath. Sorry for that, did not know about all the dependencies and requirements in the background.

Reply:
@votdev thanks, and no problem, I didn't knew about this problem either.

Reply:
FYI, I opened issue #814

Reply:
PR #815 will fix the slspath issue.